packages/grid/_modules_/grid/components/column-header-menu-icon.tsx

```diff
@@ -0,0 +1,42 @@
+import * as React from 'react';
+import IconButton from '@material-ui/core/IconButton';
+import { useGridState } from '../hooks/features/core/useGridState';
+import { useIcons } from '../hooks/utils/useIcons';
+import { ApiContext } from './api-context';
+import { ColDef } from '../models/colDef/colDef';
+
+export interface ColumnHeaderFilterIconProps {
+  column: ColDef;
+}
+
+export const ColumnHeaderMenuIcon: React.FC<ColumnHeaderFilterIconProps> = React.memo(
+  ({ column }) => {
+    const icons = useIcons();
+    const icon = React.createElement(icons.ColumnMenu!, {});
+    const apiRef = React.useContext(ApiContext);
+    const [, setGridState, forceUpdate] = useGridState(apiRef!);
+
+    const menuIconClick = React.useCallback(() => {
+      setGridState((state) => ({
+        ...state,
+        columnMenu: { open: true, field: column.field },
+        preferencePanel: { open: false },
+      }));
+      forceUpdate();
```

Comment:
I don't remember if we had a duscussion about having the `forceUpdate` being called from within the `setGridState`. It seems that we need to call it every time the `setGridState` is called.

Reply:
Not in every case, especially in useRows